server/etcdserver/api/v3rpc/grpc.go

```diff
@@ -66,8 +62,8 @@ func Server(s *etcdserver.EtcdServer, tls *tls.Config, interceptor grpc.UnarySer
 	opts = append(opts, grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(chainUnaryInterceptors...)))
 	opts = append(opts, grpc.StreamInterceptor(grpc_middleware.ChainStreamServer(chainStreamInterceptors...)))
 
-	opts = append(opts, grpc.MaxRecvMsgSize(int(s.Cfg.MaxRequestBytes+grpcOverheadBytes)))
-	opts = append(opts, grpc.MaxSendMsgSize(maxSendBytes))
+	opts = append(opts, grpc.MaxRecvMsgSize(s.Cfg.MaxRecvMsgSize))
```

Comment:
What if `MaxRequestBytes` is different from `s.Cfg.MaxRecvMsgSize`?  If you configure a big value for `s.Cfg.MaxRecvMsgSize`, i.e. 10MB, while the `MaxRequestBytes` is 1 MB, then etcdserver may still reject the request.

Reply:
Do you mean we should ensure `MaxRecvMsgSize` is larger than `MaxRequestBytes`? The internal raft request is still restricted by `MaxRequestBytes`.

Reply:
Do you mean `MaxRecvMsgSize` should be larger or equal to `MaxRequestBytes`?

Reply:
I don't see any reason to add the two parameters. 

Did you see any issue without adding MaxRecvMsgSize/MaxSendMsgSize?

Reply:
Actually, we have a feature that needs to change `MaxRecvMsgSize` to 150M. I was wondering if etcd can offer a separate config to customize it without changing `MaxRequestBytes`.  For now, when we start a cluster, it will give a warning about the `MaxRequestBytes` exceeds the recommended size.

Reply:
A couple of comments:
1. etcd is designed to serve small metadata. Why do you need to change `MaxRecvMsgSize` to 150M? Can you elaborate your use case?
2. Have you verified your case based on this PR with `MaxRecvMsgSize` being set to 150M, while keep `--max-request-bytes` unchanged?

Reply:
Any update on this?

Reply:
Sorry for the late reply.
> 1. etcd is designed to serve small metadata. Why do you need to change MaxRecvMsgSize to 150M? Can you elaborate your use case?

We will register our service into the embed etcd. See https://github.com/tikv/pd/blob/0d05fba64372408c5af9a78cd9b47b58c4925ca9/server/server.go#L277-L280. For example, the store heartbeat https://github.com/tikv/pd/blob/0d05fba64372408c5af9a78cd9b47b58c4925ca9/server/grpc_service.go#L570.When we need to recover our cluster, the heartbeat request will bring the meta of data, which could be large. We utilize the meta to decide which part of the data should be recovered.

> 2. Have you verified your case based on this PR with MaxRecvMsgSize being set to 150M, while keep --max-request-bytes unchanged?

not yet

Reply:
I would consider registering a hook to alter the grpc options as part of the config: 

`func additionalGrpcServerOptions() grpc.ServerOption[]`

Reply:
Actually: 
```
   grpcAdditionalServerOptions grpc.ServerOption[]
```
in the config would do the trick.